### ISSUE TITLE
Support polling `kqueue` file descriptors with `AsyncFd`

### ISSUE THREAD

**Issue by tvallotton** [tokio-rs/tokio#6196]
**Is your feature request related to a problem? Please describe.**
So my problem is I'm developing a runtime of my own, and I would like to make it as compatible as possible with `tokio`. My runtime has a share nothing architecture, so it really benefits from waking from the same thread. This means that I want the two runtimes to run in the same thread. The thing is, we can't have both reactors block the current thread. 

In my view the solution is simple, one reactor needs to be polled by the other. Tokio doesn't expose the file descriptor of the epoll/kqueue instance in their public API. However, it does expose a generic `AsyncFd` struct for polling user provided file descriptors. To me this means I can have my reactor be polled by tokio using `AsyncFd`, and have both runtimes share a thread. This works fine on linux polling an io-uring file descriptor (as far as I've tested), but it does not work on BSD when trying to poll a kqueue file descriptor. 

TLDR: `AsyncFd` cannot be used to poll `kqueue` file descriptors. 

**Describe the solution you'd like**

I would like the following test to pass
```rust
#[test]
fn kqueue_in_async_fd() {
        let kq = unsafe { libc::kqueue() };
        assert!(kq != -1, "{}", std::io::Error::last_os_error());

        tokio::io::unix::AsyncFd::new(kq).unwrap();
}
```
Currently it panics with: 
```
called `Result::unwrap()` on an `Err` value: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }
```
I'm not entirely sure what checks are being performed. But I would like to be able to skip these checks so I can poll that file descriptor. 

**Describe alternatives you've considered**
A good alternative would be for tokio to expose the `RawFd` of their reactor so I can poll it from my reactor. 

**Additional context**

**Comment by h33p** [comment]
You need to [attach compatible polling interest](https://github.com/memflow/mfio/blob/main/mfio/src/backend/integrations/tokio.rs#L89) to your handle. `kqueue` objects are only read-pollable.

**Comment by tvallotton** [comment]
Thank you, this solves the issue I was having.
